src/dbms/replication_client.cpp

```diff
@@ -10,25 +10,32 @@
 // licenses/APL.txt.
 
 #include "dbms/replication_client.hpp"
+#include "replication/replication_client.hpp"
 
 namespace memgraph::dbms {
 
 void StartReplicaClient(DbmsHandler &dbms_handler, replication::ReplicationClient &client) {
   // No client error, start instance level client
   auto const &endpoint = client.rpc_client_.Endpoint();
   spdlog::trace("Replication client started at: {}:{}", endpoint.address, endpoint.port);
-  client.StartFrequentCheck([&dbms_handler](std::string_view name) {
-    // Working connection, check if any database has been left behind
-    dbms_handler.ForEach([name](dbms::Database *db) {
+  client.StartFrequentCheck([&client, &dbms_handler](std::string_view name, bool reconnect) {
+    // Working connection
+    // Check if system needs restoration
+    if (reconnect) client.behind_ = true;
+#ifdef MG_ENTERPRISE
+    dbms_handler.SystemRestore(client);
+#endif
+    // Check if any database has been left behind
+    dbms_handler.ForEach([name, reconnect](dbms::DatabaseAccess db_acc) {
       // Specific database <-> replica client
-      db->storage()->repl_storage_state_.WithClient(name, [&](storage::ReplicationStorageClient *client) {
-        if (client->State() == storage::replication::ReplicaState::MAYBE_BEHIND) {
+      db_acc->storage()->repl_storage_state_.WithClient(name, [&](storage::ReplicationStorageClient *client) {
```

Comment:
If we have client, I don't think we need to refind it by name using WithClient

Reply:
Different clients.
Maybe need to change names.

Reply:
Gonna skip this for now.